main.nf

```diff
@@ -421,26 +421,27 @@ process samtools {
 
 
 /*
- * STEP 3.3 - Statistics about unmapped reads against ref genome
+ * STEP 3.3 - Statistics about mapped reads against ref genome
  */
 
-process bwa_unmapped {
+process bwa_mapped {
     tag "${input_files[0].baseName}"
-    publishDir "${params.outdir}/bwa/unmapped", mode: 'copy'
+    publishDir "${params.outdir}/bwa/mapped", mode: 'copy'
 
     input:
-    file input_files from bam_for_unmapped.collect()
+    file input_files from bam_for_mapped.collect()
+    file bai from bai_for_mapped.collect()
 
     output:
-    file 'unmapped_refgenome.txt' into bwa_unmapped
+    file 'mapped_refgenome.txt' into bwa_mapped
 
     script:
     """
     for i in $input_files
```

Comment:
Maybe you've answered this before, but why do `.collect()` and then iterate through in bash? Is it not better to just let the process run for each bam as normal? Doing it as it is currently prevents parallelisation.